### ISSUE TITLE
Transfer speeds 2x slower with 3.1.1 dialect compared to other dialects, from Windows 10 PC

### ISSUE THREAD

**Issue by bubbleguuum** [hierynomus/smbj#687]
While testing download speed of large files, I noticed that 3.1.1 dialect is 2x slower than all other dialects (including 3.0.2)
That from a Windows 10 PC, 21H2, build 19044.1466.

The config for the client is the default one with just ` .withTransactBufferSize(64 * 1024)` which is needed for 3.1.1 dialect to work at all.
Although I tested that using this same buffer size with the other dialects has no impact on download speeds. 

The file are accessed via:

```
 File file  = diskShare.openFile(absolutePath,
                    EnumSet.of(AccessMask.FILE_READ_DATA),
                    EnumSet.of(FileAttributes.FILE_ATTRIBUTE_NORMAL),
                    EnumSet.of(SMB2ShareAccess.FILE_SHARE_READ),
                    SMB2CreateDisposition.FILE_OPEN,
                    EnumSet.noneOf(SMB2CreateOptions.class))
file.read(outputStream)
;
```

The only workaround for now I've found is to disable 3.1.1 dialect in the config:

`withDialects(SMB_3_0_2, SMB_3_0, SMB_2_1, SMB_2_0_2)`

This issue does not happen when I do the same test from my Samba Server (I checked it uses 3.1.1).

**Comment by bubbleguuum** [comment]
> If you connect to a Windows 10 SMB Server, you're getting a drop in speed if you use dialect 3.1.1?

Correct. I get about 3MB/s with 3.1.1 vs 6-7MB/s with all other dialects. Only on the Windows 10 server. 


Does 3.1.1 encryption always take effect even if the config has  `withEncryptData(false)` (which is the default) ?

The Windows 10 PC and the machine running the Samba server have comparable CPUs in term of speed.

**Comment by hierynomus** [comment]
Your best bet would be to see where the slowdown is by enabling trace logging. SMBJ then logs the time it took and the time to response

**Comment by 3c71** [comment]
Old issue, but I'm facing similar issue, very slow SMBJ data transfer, and didn't find any option/buffer sizes that improves it.
Whether I disable SMB 3.x entirely or not (thus relying on SMB 2.x), transfer rate tops at 8.00MB/s (with some drops to 4.00MB/s).

If I remove this line, it drops to 2.00MB/s !?
`.withNegotiatedBufferSize()`


**What is strange is that it tops at 8 or 2 MB/s sharp without any fluctuation as if auto-limited ?**


FWIW the mobile used for testing can download file at 40+MB/s from Internet, and my PC transfer files from the same share between 40 and 50MB/s !?

From the exact same spot, using WebDav protocol tops at 12,xxMB/s and FTP protocol tops at 15.xxMB/s and both are fluctuating as expected.

EDIT: Tested an app on Play Store supporting SMB, and it transferred files at about 33MB/s, not sure which SMB library it uses, but I would expect SMBJ to be able to support such.

Either I'm doing something wrong, or there are limitations I'm not aware of.

EDIT2: Digging into buffer sizes (testing various values for .withBufferSize()), I realized that buffer size matches exactly the speed limit. 4MB buffer = 4MB/s, etc. But setting buffer size above 8MB does not actually work, negotiated protocol ends-up limited to 8MB:
`NegotiatedProtocol{dialect=SMB_3_1_1, maxTransactSize=8388608, maxReadSize=8388608, maxWriteSize=8388608}`


This is like the library will write 1 buffer every second, but no more.
